### PR TITLE
[Snyk + GH Actions] Update requirements

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Update requirements
- Updated `requirements.in` + `requirements.txt`
- Auto-generated by [snyk.yml][1]

[1]: https://github.com/gsa/catalog.data.gov/.github/workflows/snyk.yml

### CODE REVIEW

File: ckan/requirements.txt [hunk 1]
```diff
@@ -93,7 +94,7 @@ Routes==1.13
 rq==1.0
 s3transfer==0.6.0
 setuptools==65.6.3
-Shapely==1.8.5.post1
+shapely==2.0.0
```
Comment (jbrown-xentity) [inline ckan/requirements.txt:97]: This seems potentially dangerous, but tests passed so 🤷‍♂️

File: ckan/requirements.txt [hunk 2]
```diff
@@ -56,6 +56,7 @@ MarkupSafe==2.1.1
 messytables==0.15.2
 newrelic==8.5.0
 nose==1.3.7
+numpy==1.23.5
```
Comment (jbrown-xentity) [inline ckan/requirements.txt:59]: I wonder why this got added...

**Review by jbrown-xentity — APPROVED** [review]

File: ckan/requirements.txt (hunk 1 above)
Comment (nickumia-reisys) [reply ckan/requirements.txt:97]: I would think so too, but hey, +1 to the snyk script, right? 😆

File: ckan/requirements.txt (hunk 2 above)
Comment (nickumia-reisys) [reply ckan/requirements.txt:59]: It might be a new dependency for shapely?

**Comment by nickumia-reisys** [comment]
So... @jbrown-xentity While this is a good thing to do, this is technically a bad PR because there was a Snyk vulnerability that was not fixable, so it took no snyk action, but still updated the rest of the dependencies.  Inventory had a different issue, but same situation
- https://github.com/GSA/data.gov/issues/4125

Instead of only running when there is a vulnerability to fix, this will just be a daily update to dependencies which may be good to do but is not the intended feature of this script.

cc @FuhuXia @btylerburton @Jin-Sun-tts @robert-bryson

**Comment by nickumia-reisys** [comment]
Well.. it wasn't the "originally intended" feature, if we are okay with this functionality, we can leave it I guess...

**Comment by jbrown-xentity** [comment]
> So... @jbrown-xentity While this is a good thing to do, this is technically a bad PR because there was a Snyk vulnerability that was not fixable, so it took no snyk action, but still updated the rest of the dependencies. Inventory had a different issue, but same situation
> 
> * [Handle Snyk Scans with no Upgrade/Patch-ability data.gov#4125](https://github.com/GSA/data.gov/issues/4125)
> 
> Instead of only running when there is a vulnerability to fix, this will just be a daily update to dependencies which may be good to do but is not the intended feature of this script.
> 
> cc @FuhuXia @btylerburton @Jin-Sun-tts @robert-bryson

It's not a bad PR in the dangerous sense, it just didn't raise/solve all the problems the automated process was intended to. Ideally the "impossible to upgrade to fix" use case will need to be manual intervention and/or an issue to investigate if it is a vulnerability in our situation. Theoretically the script could create an issue, and/or do nothing if an issue with the same title is already created (whether closed or open). That seems like a lot of work and authentication though... 🤷‍♂️

**Comment by nickumia-reisys** [comment]
I will defer the conversation about whether this is a lot of work to after we decide whether we need a human to make a decision that a machine can not.  If a human has to decide whether this is 
>  an issue to investigate if it is a vulnerability in our situation

... then, the point of quantity of work is not a topic since a machine can't make that decision.


My only comment in that case will be:  It's an O&M Task to check this action and make sure that unfixable vulnerabilities are triaged and then we can prevent this issue from constantly failing in an unintended way.

**Pull request merged**